packages/headless/src/utils/validate-payload.ts

```diff
@@ -40,27 +41,35 @@ export const validatePayloadValue = <P>(
 };
 
 export const validateInitialState = <T extends object>(
+  engine: Engine<unknown>,
   schema: Schema<T>,
   obj: T | undefined,
   functionName: string
 ) => {
   const message = `Check the initialState of ${functionName}`;
-  return validateObject(schema, obj, message);
+  return validateObject(engine, schema, obj, message);
 };
 
 export const validateOptions = <T extends object>(
+  engine: Engine<unknown>,
   schema: Schema<T>,
   obj: Partial<T> | undefined,
   functionName: string
 ) => {
   const message = `Check the options of ${functionName}`;
-  return validateObject(schema, obj, message);
+  return validateObject(engine, schema, obj, message);
 };
 
 const validateObject = <T extends object>(
+  engine: Engine<unknown>,
   schema: Schema<T>,
   obj: T | undefined,
   message: string
 ) => {
-  return schema.validate(obj, message);
+  try {
+    return schema.validate(obj, message);
+  } catch (error) {
+    engine.logger.error(error, 'Controller initialization error');
```

Comment:
If the function name is `validateObject` (which, I assume, means validate `any` object), why does it contain the keyword `Controller` here ?